code/game/objects/items/toys.dm

```diff
@@ -1593,75 +1613,166 @@
 		input = GRAB
 	if(icon_x < ICON_SPLIT && icon_y < ICON_SPLIT)
 		input = HARM
-	render(input)
-	player_sequence += input
+
+	player_input(user, input)
+
+/obj/item/toy/intento/proc/boot()
+	say("Game starting!")
+	playsound(src, 'sound/machines/synth_yes.ogg', 50, FALSE)
+
+	state = STATE_STARTING
+	next_process = world.time + TIME_TO_BEGIN
+	START_PROCESSING(SSfastprocess, src)
+
+/obj/item/toy/intento/proc/player_input(mob/player, intent)
+	// All branches of this proc lead to us wanting to process
+	START_PROCESSING(SSfastprocess, src)
+
+	render(intent)
+
+	player_sequence += intent
 	for(var/i in 1 to player_sequence.len)
 		if(player_sequence[i] != current_sequence[i])
-			return end(user)
+			state = STATE_END_OF_GAME
+			next_process = world.time + TIME_TO_RESET_ICON
+			return
+
 	if(player_sequence.len == current_sequence.len)
 		score++
-		addtimer(CALLBACK(src, .proc/start_round, 1 SECONDS))
+
+		state = STATE_STARTING
+		next_process = world.time + TIME_TO_BEGIN
+
+
+/obj/item/toy/intento/process()
+	if(next_icon_reset && next_icon_reset <= world.time)
+		icon_state = initial(icon_state)
+		next_icon_reset = 0
+
+	if(next_process && next_process > world.time)
 		return
-	usable = TRUE
 
-/obj/item/toy/intento/proc/start()
-	say("Game starting!")
-	playsound(src, 'sound/machines/synth_yes.ogg', 50, FALSE)
-	on = TRUE
-	addtimer(CALLBACK(src, .proc/start_round, 1 SECONDS))
-
-/obj/item/toy/intento/proc/end(mob/user)
-	var/award_score = score
-	var/award_status = user.client.get_award_status(/datum/award/score/intento_score)
-	if(award_score - award_status > 0)
-		award_score -= award_status
-	user.client.give_award(/datum/award/score/intento_score, user, award_score, TRUE)
+	switch(state)
+		if(STATE_STARTING)
+			process_start()
+
+		if(STATE_DEMO)
+			process_demo()
+
+		if(STATE_END_OF_GAME)
+			var/mob/living/holder = null
+			if(isliving(loc))
+				holder = loc
+
+			process_end(holder)
+
+		if(STATE_RETALIATION)
+			process_retaliation()
+
+	if(!next_process && !next_icon_reset)
+		return PROCESS_KILL
+
+/obj/item/toy/intento/proc/process_start()
+	player_sequence.Cut()
+
+	current_sequence += pick(list(HELP, DISARM, GRAB, HARM))
+
+	state = STATE_DEMO
+	next_process = world.time
+	index = 1
+
+/obj/item/toy/intento/proc/process_demo()
+	if(index > length(current_sequence))
+		state = STATE_AWAITING_PLAYER_INPUT
+		next_process = 0
+		return
+
+	var/intent = current_sequence[index]
+	render(intent)
+
+	index += 1
+	next_process = world.time + TIME_PER_DEMO_STEP
+
+/obj/item/toy/intento/proc/process_end(mob/user)
+	if(user)
+		var/award_score = score
+		var/award_status = user.client.get_award_status(/datum/award/score/intento_score)
+		if(award_score - award_status > 0)
+			award_score -= award_status
+		user.client.give_award(/datum/award/score/intento_score, user, award_score, TRUE)
```

Comment:
```suggestion
		user.client.give_award(/datum/award/score/intento_score, user, award_score)
```

I can't find a TRUE being needed in any definition of give_award. Correct me if I'm wrong.